### PR TITLE
[AMD-ZENDNN] Modifications to support ZenDNN mempool optimizations

### PR DESCRIPTION
**Pull request by aadwived**
- ZenDNN mempool optimizations utilizes pre-allocated buffers with TF_Tensor.
- Hence, amount of allocated TF_Tensor bytes can be greater than or equal to requested bytes by the Op execution.

Authors:
  Aakar Dwivedi ( aakar.dwivedi@amd.com )
  Avinash-Chandra Pandey ( avinash-chandra.pandey@amd.com )
  Chandra Kumar Ramasamy ( chandrakumar.ramasamy@amd.com )
  Savan Anadani ( savan.anadani@amd.com )

Signed-off-by: Aakar Dwivedi <Aakar.Dwivedi@amd.com>

### CODE REVIEW

File: tensorflow/python/lib/core/ndarray_tensor.cc [hunk 1]
```diff
@@ -473,8 +474,20 @@ Status TF_TensorToPyArray(Safe_TF_TensorPtr tensor, PyObject** out_ndarray) {
...
+              TF_TensorByteSize(tensor.get()))) {
+    return errors::Internal("ndarray was ", PyArray_NBYTES(py_array),
+                            " bytes but TF_Tensor was ",
+                            TF_TensorByteSize(tensor.get()), " bytes");
+  }
+  // Default TF implementation compares ndarray bytes with TF_Tensor bytes
+  // which is allocated using allocate_output API for fixed size.
+  // ZenDNN mempool optimization utilizes pre-allocated buffers with TF_Tensor.
+  // Hence, amount of allocated TF_Tensor bytes can be greater than or equal
+  // to requested bytes by the Op execution.
```
Comment (penpornk) [inline tensorflow/python/lib/core/ndarray_tensor.cc:488]: Nit: It wasn't obvious to me that "can be >=" in here means "can't be <" so I think we should make the wording stricter.
```suggestion
  // Hence, the amount of allocated TF_Tensor bytes is always greater than or
  // equal to the number of bytes requested by the op execution.
```

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR!

File: tensorflow/python/lib/core/ndarray_tensor.cc (hunk 1 above)
Comment (aadwived) [reply tensorflow/python/lib/core/ndarray_tensor.cc]: Resolved. Thank you for pointing out the issue.

**Review by penpornk — APPROVED** [review]
Thank you for the quick changes!

**Pull request merged**